From 77d6b26f1de66a549c260cb3db6bcbba46ce9b3b Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 08 Apr 2022 08:55:58 +0300
Subject: [PATCH] [SW-79695] gaudi3: update revokation bits location

As per the updated efuse bits definition for gaudi3, we can
remove alternate location (spi) for keys revokation bits.

Revokation bits will only be read from efuse.

Change-Id: I79b03f4b7df16a518272a92aa36e0fd191663cf5
---

diff --git a/arch/gaudi3/arc/dev_efuse.h b/arch/gaudi3/arc/dev_efuse.h
index cb9e5a7..ac4a1c1 100644
--- a/arch/gaudi3/arc/dev_efuse.h
+++ b/arch/gaudi3/arc/dev_efuse.h
@@ -19,15 +19,16 @@
 #define EFUSE_SYSCFG_ENABLE_TPM_MASK			BIT(11)
 #define EFUSE_SYSCFG_FW_VERSION_LOCATION		7
 #define EFUSE_SYSCFG_FW_VERSION_LOCATION_MASK		0x3
-#define EFUSE_SYSCFG_REVOKATION_LOCATION		7
-#define EFUSE_SYSCFG_REVOKATION_LOCATION_MASK		0x3
 
 #define EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI		BIT(8)
-#define EFUSE_SYSCFG_REVOKATION_LOCATION_SPI		BIT(8)
 
 #define EFUSE_SYSCFG_EFUSE_LOCATION_MASK		BIT(25)
 #define EFUSE_SYSCFG_SKU_MASK				BIT(26)
 
+/* Following bitmaps are used in common code, hence maintained for compilation */
+#define EFUSE_SYSCFG_REVOKATION_LOCATION		0
+#define EFUSE_SYSCFG_REVOKATION_LOCATION_MASK		0x0
+
 /*
  * As per recommendation from Security Architect, if FUSE is not available,
  * card should not continue in the secure boot mode. Hence the default
diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 144fec3..b6f5cdf 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -177,27 +177,12 @@
 
 void security_load_revoke_bits(void)
 {
-	uint32_t key_revoke = 0;
-	enum security_location revoke_location;
-
-	revoke_location = ((arcboot_rom_global.config_fuse >>
-			   EFUSE_SYSCFG_REVOKATION_LOCATION) &
-			   EFUSE_SYSCFG_REVOKATION_LOCATION_MASK);
+	uint32_t key_revoke = arcboot_rom_global.config_fuse & EFUSE_SYSCFG_REVOKE_KEY_MASK;
 
 	arcboot_rom_status_update(PROGRESS_REVOKE_LOC);
+	/* REVOKE BITS LOCATION: EFUSE */
 	arcboot_rom_status_update_location(PROGRESS_REVOKE_LOC_0,
-					   revoke_location);
-
-	switch (revoke_location) {
-	/* TODO: Add other locations if required for gaudi3 */
-	/* TODO: Confirm enumeration values for gaudi3 */
-	case ASIC_EFUSE:
-		key_revoke = arcboot_rom_global.config_fuse;
-		break;
-	default:
-		return;
-	}
-	key_revoke &= EFUSE_SYSCFG_REVOKE_KEY_MASK;
+					   ASIC_EFUSE);
 
 	/* store key_revoke map in PSOC */
 	regmap_writel(key_revoke, mmSECURITY_KEY_REVOKE);
